runtime/protocol/http/common/src/aws/smithy/kotlin/runtime/http/middleware/Retry.kt

```diff
@@ -41,6 +53,9 @@ class Retry<O>(
                 // Deep copy the request because later middlewares (e.g., signing) mutate it
                 val requestCopy = request.deepCopy()
 
+                // we don't know max attempts, it comes from the strategy and setting ttl would never be accurate
+                // set attempt which is the only additional metadata we know
+                requestCopy.subject.header(AMZ_SDK_REQUEST_HEADER, "attempt=$attempt")
```

Comment:
**Comment**: We could expose this relatively simply:

```kotlin
interface RetryStrategy {
    val options: RetryOptions
    suspend fun <R> retry(policy: RetryPolicy<R>, block: suspend () -> R): Outcome<R>
}

interface RetryOptions {
    val maxAttempts: Int? = null
}

data class StandardRetryStrategyOptions(override val maxAttempts: Int) : RetryOptions {
    ...
}
```

Keeping the `RetryOptions` member null allows exposing a relatively common retry option without forcing retry strategies to use it internally (even though our _only_ retry strategy _does_ implement it).

What do you think about this?

Reply:
Yeah we could. It is called out in SEP that if the value is unknown or not reliable to omit it. Depends on how important we really think it is to include. My vote is not all that important (I almost left out `AMZ_SDK_REQUEST_HEADER` all together)

Reply:
I'd like to include it if we could. It may provide useful information to services (IDK if any of them make use of it at this point).